tools/restatectl/src/commands/metadata_server/nodes.rs

```diff
@@ -133,5 +139,54 @@ async fn remove_node(
         }
     }
 
+    if one_node_failed {
+        return Err(anyhow::anyhow!(
+            "Failed removing some nodes from the metadata cluster"
+        ));
+    }
+
     Ok(())
 }
+
+async fn try_remove_node(
+    node_to_remove: PlainNodeId,
+    metadata_node_set: &HashMap<PlainNodeId, &NodeConfig>,
+    connection_info: &ConnectionInfo,
+) -> Result<(), HashMap<PlainNodeId, Status>> {
+    assert!(
+        !metadata_node_set.is_empty(),
+        "metadata cluster is not empty"
+    );
+
+    let mut errors = HashMap::new();
+    // todo try to figure out who's the current leader and directly reach out to the leader
+    //  based on the metadata server status call
+    for (server_id, node_config) in metadata_node_set.iter() {
+        let channel = match connection_info.connect(&node_config.address).await {
+            Ok(channel) => channel,
+            Err(err) => {
+                errors.insert(*server_id, Status::unavailable(err.to_string()));
+                continue;
+            }
+        };
+        let mut client = MetadataServerSvcClient::new(channel);
+
+        match client
+            .remove_node(RemoveNodeRequest {
+                plain_node_id: u32::from(node_to_remove),
+                created_at_millis: None,
+            })
+            .await
+        {
+            Ok(_response) => {
+                return Ok(());
+            }
+            Err(err) => {
+                debug!(%err, "Failed removing node from the metadata cluster. Trying different metadata server.");
+                errors.insert(*server_id, err);
```

Comment:
we need to make this idempotent. we may get an error but it actually succeeded. if the error is 'N is not a member' then we should consider it success. 
<img width="956" height="262" alt="image" src="https://github.com/user-attachments/assets/e19c92c5-357c-41c3-9efd-f371ac266e28" />